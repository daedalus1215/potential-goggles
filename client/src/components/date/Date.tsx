import cn from 'classnames';
import styles from './Date.module.css';

type DateInterface = ({ date?: string, setDate: (date: string) => void, classNames?: string });

export const Date: React.FC<DateInterface> = ({ date, setDate, classNames }) => <input
    className={cn(styles.date, classNames)}
    type='date'
    name='date'
    id='date'
    value={date}
    onChange={e => setDate(e?.target.value)}
/>

