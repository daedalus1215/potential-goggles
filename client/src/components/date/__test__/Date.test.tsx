import { render } from "@testing-library/react";
import { Date } from "../Date";

describe('client/src/components/date/__test__/Date.test.tsx', () => {
    it('should render', () => {
        // Arrange & Act
        const target = render(<Date date='2023-12-12' setDate={jest.fn()} />)

        // Assert
        expect(target.findByTestId('date'));
        expect(target.findByTestId("date")).toMatchSnapshot();
    });

    // @TODO: Do onclick test
});