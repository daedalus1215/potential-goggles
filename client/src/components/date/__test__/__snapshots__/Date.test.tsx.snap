// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/src/components/date/__test__/Date.test.tsx should render 1`] = `Promise {}`;

exports[`client/src/components/date/__test__/Date.test.tsx should... 1`] = `Promise {}`;
